dulwich/repo.py

```diff
@@ -1272,6 +1273,13 @@ def _write_reflog(
                 + b"\n"
             )
 
+    def _reflog_path(self, ref: bytes) -> str:
+        assert not ref.startswith((b"main-worktree/", b"worktrees/")), (
+            "special paths are not supported"
+        )
```

Comment:
There are special refs, like `main-worktree/` and `worktrees/` that gives you access to per-worktree refs of the main worktree, and per-worktree refs of the linked worktrees respectively.

I have not implemented those.